onedal/_device_offload.py

```diff
@@ -194,17 +187,29 @@ def support_usm_ndarray(freefunc=False, queue_param=True):
 
     def decorator(func):
         def wrapper_impl(obj, *args, **kwargs):
-            usm_iface = _extract_usm_iface(*args, **kwargs)
+            if len(args) == 0 and len(kwargs) == 0:
+                return _run_on_device(func, obj, *args, **kwargs)
+            data = (*args, *kwargs.values())
```

Comment:
where is ```data``` used in this? having trouble finding it.

Reply:
It used on the lines 199, 207 and 209. I will think either `args[0]` or `data[0]`